routers/web/admin/admin.go

```diff
@@ -133,12 +136,22 @@ func DashboardPost(ctx *context.Context) {
 
 	// Run operation.
 	if form.Op != "" {
-		task := cron.GetTask(form.Op)
-		if task != nil {
-			go task.RunWithUser(ctx.Doer, nil)
-			ctx.Flash.Success(ctx.Tr("admin.dashboard.task.started", ctx.Tr("admin.dashboard."+form.Op)))
-		} else {
-			ctx.Flash.Error(ctx.Tr("admin.dashboard.task.unknown", form.Op))
+		switch form.Op {
+		case "sync_repo_branches":
+			go func() {
+				if err := repo_service.AddAllRepoBranchesToSyncQueue(graceful.GetManager().ShutdownContext(), ctx.Doer.ID); err != nil {
```

Comment:
If the button is clicked multiple times, then the queue will have many duplicate sync tasks?

Reply:
> If the button is clicked multiple times, then the queue will have many duplicate sync tasks?

It's a unique queue, so for one repository, there is no duplicated task at the same time.